src/lib/structures/Josh.ts

```diff
@@ -123,6 +125,11 @@ export class Josh<StoredValue = unknown> {
    * ```javascript
    * josh.use(new MyMiddleware());
    * ```
+   *
+   * @example
+   * ```javascript
+   * josh.use(new MyMiddleware().setContext({ data: 'data' }));
```

Comment:
Move setting of context to the constructor perhaps (discussed in dsc)

Reply:
"Josh.Options#autoEnsure, if they have the option there it adds auto ensure and so on for other built-in middleware" - @RealShadowNova